test/Microsoft.AspNet.Server.KestrelTests/SocketOutputTests.cs

```diff
@@ -192,6 +192,83 @@ public void WritesDontCompleteImmediatelyWhenTooManyBytesIncludingNonImmediateAr
             }
         }
 
+        [Fact]
+        public void WritesDontGetCompletedTooQuickly()
```

Comment:
You'll probably have to skip this test on Mono (https://github.com/aspnet/KestrelHttpServer/pull/373).